.github/workflows/dev.yml

```diff
@@ -0,0 +1,33 @@
+name: Dev branch CI
+
+on:
+  pull_request:
+    branches:
+      - dev
+
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up JDK 17 for x64
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          architecture: x64
+
+      - name: Cache Maven dependencies
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-maven-
+
+      - name: Clean and build
+        run: mvn -B -e clean install -DskipTests
```

Comment:
By running clean every time, isn't the cache being deleted?

Reply:
No. I run the CI again and the dependences not be downloaded. Was used using the cache.